src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Model/EitherTest.cs

```diff
@@ -93,6 +93,165 @@ public void FoldLeft_Right()
 
             Assert.Equal("a default", result);
         }
+
+        [Fact]
+        public void OnSuccess()
+        {
+            var either =
+                new Either<int, string>("either1")
+                    .OnSuccess(either1Result => either1Result + " either2");
+            
+            Assert.True(either.IsRight);
+            Assert.Equal("either1 either2", either.Right);
+        }
+
+        [Fact]
+        public void OnSuccess_Left_PriorFailure()
+        {
+            var eitherResults = new List<string>();
+            
+            var either = 
+                new Either<int, string>(500)
+                    .OnSuccess(previousResult =>
+                    {
+                        eitherResults.Add(previousResult + " either2");
+                        return previousResult + "either2";
+                    });
+            
+            Assert.True(either.IsLeft);
+            Assert.Equal(500, either.Left);
+            
+            Assert.Empty(eitherResults);
+        }
+        
+        [Fact]
+        public void OnSuccess_NoArg()
+        {
+            var either =
+                new Either<int, string>("either1")
+                    .OnSuccess(() => "either2");
+            
+            Assert.True(either.IsRight);
+            Assert.Equal("either2", either.Right);
+        }
+
+        [Fact]
+        public void OnSuccess_NoArg_Left_PriorFailure()
+        {
+            var eitherResults = new List<string>();
+            
+            var either = 
+                new Either<int, string>(500)
+                    .OnSuccess(() =>
+                    {
+                        eitherResults.Add("either2");
+                        return "either2";
+                    });
+            
+            Assert.True(either.IsLeft);
+            Assert.Equal(500, either.Left);
+            
+            Assert.Empty(eitherResults);
+        }
+
+        [Fact]
+        public async Task OnSuccess_WithEitherTask()
```

Comment:
These are actually in the correct place because they are methods on the Either class itself, rather than the extension methods in EitherTaskExtensions.  They operate from an initial Either, rather than an initial Task<Either>, but it's confusing as they then deal with Task<Either<?,?>> method params.

Reply:
Ah I see, that is a little confusing, but fair enough!